packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java

```diff
@@ -124,27 +124,16 @@ public void registerEndpoints(URL openApiResource) {
         endpointBeans
                 .putAll(context.getBeansWithAnnotation(BrowserCallable.class));
 
-        if (endpointRegistry.isEmpty() && !endpointBeans.isEmpty()) {
-            LOGGER.debug("No endpoints found in openapi.json:"
-                    + " registering all endpoints found using the Spring context");
+        var currentEndpointNames = endpointBeans.values().stream()
+                .map(endpointRegistry::registerEndpoint)
+                .collect(Collectors.toSet());
```

Comment:
As the Vaadin endpoint names are kept as all-lowercase, an extra mapping is needed, so that the `retainAll` doesn't remove all the endpoints:
```suggestion
        var currentEndpointNames = endpointBeans.values().stream()
                .map(endpointRegistry::registerEndpoint)
                .map(String::toLowerCase)
                .collect(Collectors.toSet());
```